The "Duplicate object" action with "update incoming reference" should only update changeable feature

When duplicating a Object with the option "update incomming reference" the non changeable incomming reference should not be updated. If it does, we might end up trying to modify _Unmodifiable_ collection.

```
java.util.concurrent.ExecutionException: java.lang.UnsupportedOperationException
    at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[na:na]
    at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191) ~[na:na]
    ...
Caused by: java.lang.UnsupportedOperationException: null
    at org.eclipse.emf.common.util.BasicEList$UnmodifiableEList.add(BasicEList.java:907) ~[org.eclipse.emf.common-2.31.0.jar:na]
    at org.eclipse.sirius.web.application.views.explorer.services.DefaultObjectDuplicator.lambda$updateIncomingReferences$4(DefaultObjectDuplicator.java:106) ~[sirius-web-application-2025.8.10-SNAPSHOT.jar:2025.8.10-SNAPSHOT]
    at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183) ~[na:na]
    at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179) ~[na:na]
    at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625) ~[na:na]
    at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762) ~[na:na]
    at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276) ~[na:na]
    at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[na:na]
```

--- a/CHANGELOG.adoc
+++ b/CHANGELOG.adoc
@@ -26,6 +26,7 @@
 === Bug fixes
 
 - https://github.com/eclipse-sirius/sirius-web/issues/5491[#5491] [test] Add missing module in tsconfig for the playwright package
+- https://github.com/eclipse-sirius/sirius-web/issues/5627[#5627] [sirius-web] Fix the duplication of an element with incoming non changeable references
 
 
 === New Features
@@ -53,6 +54,7 @@
 - [ADR-206] Add support for importing documents with read-only permissions
 - [ADR-207] Make React contexts extensible
 - [ADR-208] Support edge type in edge style
+
 
 === Deprecation warning
 
@@ -162,7 +164,6 @@ This occured for the tree filter menu in the Explorer when a table was opened, a
 - https://github.com/eclipse-sirius/sirius-web/issues/5593[#5593] [diagram] Restore position dependant rotation for `imageNodeStyle`
 - https://github.com/eclipse-sirius/sirius-web/issues/4171[#4171] [sirius-web] Fix the display of tabs for representations with very long names, which could hide the "close tab" button (and the icon)
 - https://github.com/eclipse-sirius/sirius-web/issues/5614[#5614] [sirius-web] Add support for "pin/unpin" to all right side panels
-
 
 
 === New Features

--- a/packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/views/explorer/services/DefaultObjectDuplicator.java
+++ b/packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/views/explorer/services/DefaultObjectDuplicator.java
@@ -93,7 +93,10 @@ public class DefaultObjectDuplicator implements IDefaultObjectDuplicator {
                 .flatMap(eCrossReferenceAdapter -> eCrossReferenceAdapter.getInverseReferences(objectToDuplicate).stream())
                 .filter(setting -> {
                     boolean canFeatureBeSet = false;
-                    if (setting.getEStructuralFeature() instanceof EReference eReference && eReference.isMany() && !eReference.isContainment()) {
+                    if (setting.getEStructuralFeature() instanceof EReference eReference
+                            && eReference.isMany()
+                            && !eReference.isContainment()
+                            && eReference.isChangeable()) {
                         List<?> list = (List<?>) setting.getEObject().eGet(eReference);
                         int upperBound = eReference.getUpperBound();
                         canFeatureBeSet = upperBound == -1 || (upperBound > 0 && list.size() < upperBound);

